packages/helpers/convertPopperConfig/src/convertPopperConfig.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @typedef {{modifiers: {}, positionFixed:boolean, placement:string }} Popper1Cfg
+ * @typedef {import('@popperjs/core/lib/popper').Options} Popper2Cfg
+ */
+
+/**
+ * @param {Popper1Cfg|Popper2Cfg} pCfg
+ * @returns {boolean}
+ */
+function isPopper2Cfg(pCfg) {
+  // @ts-ignore
+  return Array.isArray(pCfg.modifiers) || pCfg.strategy || pCfg.onFirstUpdate || pCfg.afterWrite;
+}
+
+const methodMap = {
+  onFirstUpdate: 'onCreate',
+  afterWrite: 'onUpdate',
+  update: 'scheduleUpdate', // but returns Promise
+  forceUpdate: 'update',
+};
+
+/**
+ * Converts from Popper1 config to Popper2 config
+ * @param {Popper1Cfg} p1Cfg
+ * @returns {Popper2Cfg}
+ */
+export function convertPopperConfig(p1Cfg) {
+  if (isPopper2Cfg(p1Cfg)) {
+    // @ts-ignore
+    return p1Cfg;
+  }
+  const p2Cfg = /** @type {Popper2Cfg} */ ({});
+  Object.keys(p1Cfg).forEach(option => {
+    if (option === 'modifiers') {
+      /** @type {object[]} */
+      const modifiers = Object.keys(p1Cfg.modifiers).map(k => {
+        const result = {};
+        result.name = k;
+        result.options = p1Cfg.modifiers[k];
+        if ('enabled' in p1Cfg.modifiers[k]) {
+          result.enabled = p1Cfg.modifiers[k].enabled;
+          delete result.options.enabled;
+        }
+        if (typeof result.options.offset === 'string') {
+          result.options.offset = result.options.offset
+            .split(',')
+            .map((/** @type {string} */ c) => Number(c.replace('px', '')));
+        }
+        return result;
+      });
+      p2Cfg[option] = modifiers;
+    } else if (option === 'positionFixed') {
+      p2Cfg.strategy = p1Cfg.positionFixed === true ? 'fixed' : 'absolute';
+    } else if (option === 'scheduleUpdate') {
+      // @ts-expect-error
```

Comment:
Maybe add a small explanation. I assume this is because of the return value`p1Cfg[option]()` not being typed because it uses array accessor to get the method which cannot be typed?

Reply:
Ah I see now, it is because in Popper 1 you used to be able to hook into every update by hooking into scheduleUpdate, but in Popper 2 you can only hook into onFirstUpdate. The recommendation if you need to hook into every update instead of only the first time, is to make your own custom modifier https://popper.js.org/docs/v2/modifiers/#custom-modifiers and use the `fn` callback.

So I think we should remove this part, since adding an update property doesn't anything, to migrate this from popper 1 you will need to make your own modifier, and we cannot "auto convert" that for our users.

Reply:
Could very well be. I will not have a deeper look till next Monday, but feel free to adjust were needed :)